system/Debug/Toolbar.php

```diff
@@ -547,4 +543,34 @@ protected function format(string $data, string $format = 'html'): string
 
         return $output;
     }
+
+    /**
+     * Determine if the toolbar should be disabled based on the request headers.
+     *
+     * This method allows checking both the presence of headers and their expected values.
+     * Useful for AJAX, HTMX, Unpoly, Turbo, etc., where partial HTML responses are expected.
+     *
+     * @return bool True if any header condition matches; false otherwise.
+     */
+    private function shouldDisableToolbar(IncomingRequest $request): bool
+    {
+        foreach ($this->config->disableOnHeaders as $headerName => $expectedValue) {
```

Comment:
We should probably fall back to an empty array in case the config file wasn't updated.